src/binary-reader.c

```diff
@@ -1694,12 +1705,15 @@ static void read_custom_section(Context* ctx, uint32_t section_size) {
     uint32_t i, num_relocs, section;
     in_u32_leb128(ctx, &section, "section");
     in_u32_leb128(ctx, &num_relocs, "relocation count count");
-    CALLBACK(on_reloc_count, num_relocs);
+    WASM_ZERO_MEMORY(section_name);
+    if (section == WASM_BINARY_SECTION_CUSTOM)
+      in_str(ctx, &section_name, "section name");
+    CALLBACK(on_reloc_count, num_relocs, section, section_name);
```

Comment:
Is it intended to encode the section string after num_relocs? It would seem more natural to encode the section string immediately after the section code, which would also keep num_relocs immediately adjacent to the reloc array.